clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/lifecycle/LaunchFailureNotificationAgent.java

```diff
@@ -146,13 +151,29 @@ public void run() {
     }
   }
 
-  private void handleMessage(NotificationMessage notificationMessage) {
-    // TODO-AJ This should eventually tag the ASG with an alert
+  private static void handleMessage(ServerGroupTagger serverGroupTagger, NotificationMessage notificationMessage) {
     log.info(
       "Failure to launch instance (asgName: {}, reason: {})",
       notificationMessage.autoScalingGroupName,
       notificationMessage.statusMessage
     );
+
+    Matcher sqsMatcher = ARN_PATTERN.matcher(notificationMessage.autoScalingGroupARN);
+    if (!sqsMatcher.matches()) {
+      throw new IllegalArgumentException(notificationMessage.autoScalingGroupARN + " is not a valid ARN");
+    }
+
+    String region = sqsMatcher.group(1);
+    String accountId = sqsMatcher.group(2);
+
+    serverGroupTagger.alert(
+      AmazonCloudProvider.ID,
+      accountId,
+      region,
+      notificationMessage.autoScalingGroupName,
+      notificationMessage.event,
```

Comment:
What does the `event` here look like? is it a UUID or a static string?

If it's dynamic, it might make life easier when handling cleanup if it's static, e.g. `launch_failure`.